pkg/doc/vc/crypto/crypto.go

```diff
@@ -181,7 +181,19 @@ func (c *Crypto) SignCredential(dataProfile *vcprofile.DataProfile, vc *verifiab
 func (c *Crypto) getSigner(dataProfile *vcprofile.DataProfile, opts *signingOpts) (signer, string, error) {
 	switch {
 	case opts.VerificationMethod != "":
+		did, err := getDIDFromKeyID(opts.VerificationMethod)
+		if err != nil {
+			return nil, "", err
+		}
+
+		// if the verification method DID is added to profile externally, then fetch the private
+		// key from profile
+		if did == dataProfile.DID && dataProfile.DIDPrivateKey != "" {
```

Comment:
this may not work when someone passes verification method through options and doesn't want to use DID/privatekey from profile